app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -217,6 +213,12 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
       .setOnValueChanged { button, value -> button.showIfTrue(value.isNotNullOrEmpty()) }
       .hide()
 
+  fun addRedoButton(clickHandler: () -> Unit) =
+    addButton(ButtonAction.REDO)
+      .setOnClickListener { clickHandler() }
+      .setOnValueChanged { button, value -> button.showIfTrue(value.isNotNullOrEmpty()) }
```

Comment:
We should also add logic to hide the button if redo stack is empty.

Reply:
Done